streams/src/main/java/org/apache/kafka/streams/processor/internals/KafkaStreamingPartitionAssignor.java

```diff
@@ -183,6 +317,54 @@ public Subscription subscription(Set<String> topics) {
             }
         }
 
+        // If ZK is specified, get the tasks for each state topic and validate the topic partitions
+        if (zkClient != null) {
+
+            for (Map.Entry<String, Set<TaskId>> entry : tasksInfo.tasksForState.entrySet()) {
+                String topic = entry.getKey() + ProcessorStateManager.STATE_CHANGELOG_TOPIC_SUFFIX;
+                int numTasks = entry.getValue().size();
```

Comment:
The number of change log partitions may not be equal to the number of tasks.
It is the max value of taskId.partition, isn't it?